clang/lib/3C/ConstraintBuilder.cpp

```diff
@@ -398,6 +399,47 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
   TypeVarInfo &TVInfo;
 };
 
+class PtrToStructDef : public RecursiveASTVisitor<PtrToStructDef> {
+  public:
+    explicit PtrToStructDef(TypedefDecl *TDT) : TDT(TDT) {}
+
+    bool VisitPointerType(clang::PointerType *PT) {
+      ispointer = true;
+      return true;
+    }
```

Comment:
The thing I'm concerned about, per my comment above, is that you could see a pointer and a struct def, separately, rather than specifically a pointer to a struct def, and still return true.